runtime/lua/vim/highlight.lua

```diff
@@ -28,16 +28,30 @@ end
 --- use from init.vim via
 ---   au TextYankPost * lua require'vim.highlight'.on_yank()
 --- customize highlight group and timeout via
----   au TextYankPost * lua require'vim.highlight'.on_yank("IncSearch", 500)
+---   au TextYankPost * lua require'vim.highlight'.on_yank({higroup="IncSearch", timeout=250})
+--- customize conditions (do not highlight a visual selection) via
+---   au TextYankPost * lua require'vim.highlight'.on_yank({visual=false})
 ---
 -- @param higroup highlight group for yanked region
 -- @param timeout time in ms before highlight is cleared
--- @param event event structure
-function highlight.on_yank(higroup, timeout, event)
-  event = event or vim.v.event
+-- @param opts dictionary with options controlling the highlight:
+--              - higroup highlight group for yanked region
+--              - timeout time in ms before highlight is cleared
+--              - macro (bool) highlight when executing macro (default false)
+--              - visual (bool) highlight when yanking visual selection (default true)
+--              - event  (struct) event structure (default vim.v.event)
+function highlight.on_yank(opts)
+  opts = opts or {}
+  local event = opts.event or vim.v.event
+  local macro = opts.macro or false
+  local visual = (opts.visual ~= false)
```

Comment:
Maybe `macro_enabled` and `visual_enabled`? It's unclear if it's the setting or if it's the status here.

Reply:
`on_visual` and `on_macro`?

Reply:
Ah, I just meant the local variables, not the dictionary keys -- I think they made sense in the dictionary, but not in the function... i don't know if that's just me being weird though.

Reply:
I think they should have the same name, since the local variable (should) inherit from the option.

Reply:
👍 makes sense to me.

Reply:
`on_*` probably works, I'll think a bit if we can get a better name for that.